Remove unused preset-env dependencies

alecslupu opened the pull request:
<!--
NOTE: We are in the middle of a big redesign of the frontend.
That could mean that your PR will not get through on the next few months.
Please see https://github.com/decidim/decidim/discussions/9512
-->

#### :tophat: What? Why?
While working on #10727 i have noticed the pipeline failed, due to some babel configuration as per below stacktrace: 

Since @babel/preset-env has been updated to 7.22.0, it has removed the dependency to `@babel/plugin-proposal-private-property-in-object`. 

Since I have added `@babel/plugin-proposal-private-property-in-object`, `@babel/plugin-proposal-private-methods` and `@babel/plugin-proposal-class-properties` back in #8670, and I do not remember why, i am removing it in the current PR. 

#### Testing
Make sure the pipeline is green.

:hearts: Thank you!

andreslucena commented:
@alecslupu I'm not able to reproduce the error locally by following the instructions, but just seeing the changes (and specially the pipeline result 😄) it wouldn't hurt. I had just two doubts: 

1. In the [plugins' doc page](https://babeljs.io/docs/babel-plugin-transform-private-property-in-object) it says: 

> NOTE: This plugin is included in @babel/preset-env, in [ES2022](https://github.com/tc39/proposals/blob/master/finished-proposals.md)

So, I understand that we could remove it altogether from our [configuration](https://github.com/decidim/decidim/blob/fa5c525550ccd930c9f7a6e59980098f7fefa7ef/babel.config.json#L24) if we update `@babel/preset-env`? 

2. I've seen that this rule was introduced on #8670, but as far as I remember we are not using this new syntax. Do you remember/know why we introduced this babel plugin?

andreslucena commented:
FYI, I've removed the unfilled section from the PR template

alecslupu commented:
> @alecslupu I'm not able to reproduce the error locally by following the instructions, but just seeing the changes (and specially the pipeline result smile) it wouldn't hurt. I had just two doubts:
> 
>     1. In the [plugins' doc page](https://babeljs.io/docs/babel-plugin-transform-private-property-in-object) it says:
> 
> 
> > NOTE: This plugin is included in @babel/preset-env, in [ES2022](https://github.com/tc39/proposals/blob/master/finished-proposals.md)
> 
> So, I understand that we could remove it altogether from our [configuration](https://github.com/decidim/decidim/blob/fa5c525550ccd930c9f7a6e59980098f7fefa7ef/babel.config.json#L24) if we update `@babel/preset-env`?
> 
>     2. I've seen that this rule was introduced on [Update npm dependencies #8670](https://github.com/decidim/decidim/pull/8670), but as far as I remember we are not using this new syntax. Do you remember/know why we introduced this babel plugin?

1. I would say that if we upgrade the preset env maybe we will fix this issue. I would not try to do it in this PR, as it is currently blocking all the dev activities, and i see a potential of having unforeseen issues ...
2. I do not remember why i have added that. If i remember right, January 2022 was around a rails upgrade.

alecslupu commented:
I confirm that if we upgrade the preset env and we remove that config entry, the compilation process is successful.

ahukkanen reviewed (approved):
I tested this approach in another repository where we had the same issue and it seems to have fixed the problem.

Also, I can see that the pipeline is green in this PR so it should be fine to merge.

I would like to have some kind of guidance also for implementors and module developers facing this issue once we can release the next versions, as it will take some time from now as we just released. But I am unsure what would be the correct place for such documentation, a GitHub issue maybe?

Anyways, regarding this PR, everything seems to be fine.

andreslucena reviewed (approved):
LGTM 👍🏽

ahukkanen commented:
@andreslucena I'll leave the final merge for you since you were already looking into this. Just to add some confirmation to the issue that this seems to be fixing the problem.

andreslucena commented:
> I would like to have some kind of guidance also for implementors and module developers facing this issue once we can release the next versions, as it will take some time from now as we just released. But I am unsure what would be the correct place for such documentation, a GitHub issue maybe?

Yes, a GH issue seems to be the place, probably referencing this PR

ahukkanen commented:
> Yes, a GH issue seems to be the place, probably referencing this PR

Created #10918 explaining the workaround.

Feel free to edit if it is not clear enough.

andreslucena commented:
@alecslupu should we backport this one to v0.26 too?
